src/App.js

```diff
@@ -1,24 +1,47 @@
-import logo from './logo.svg';
-import './App.css';
+import * as React from "react";
+import Product from "../src/components/Product";
+import Box from "@mui/material/Box";
+import Grid from "@mui/material/Grid";
+import axios from "axios";
 
 function App() {
+  const [products, setProducts] = React.useState([]);
+  const getAllProducts = () => {
+    axios
+      .get(`http://localhost:3000/`)
+      .then((res) => {
+        setProducts(res.data.product);
+      })
+      .catch((error) => {
+        console.log(`error`);
+      });
+  };
+  React.useEffect(() => {
+    getAllProducts();
+  }, []);
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
+    <>
+      <Box sx={{ flexGrow: 1 }}>
+        <Grid
+          container
+          spacing={{ xs: 2, sm: 4, md: 3 }}
+          columns={{ xs: 4, sm: 8, md: 12 }}
         >
-          Learn React
-        </a>
-      </header>
-    </div>
+          {products.map((product, index) => (
+            <Grid item xs={2} sm={4} md={4} key={index}>
```

Comment:
As a general rule, it's best if we don't give index alone as a key. ( I think React discourages this as well )
So let's put product name along with index